src/libraries/Microsoft.Extensions.FileSystemGlobbing/tests/PatternMatchingTests.cs

```diff
@@ -468,5 +469,93 @@ public void StemMatchingWithParentDir()
         // exclude: **/.*/**
         // exclude: node_modules/*
         // exclude: **/.cs
+
```

Comment:
All of the existing tests for unordered filtering should also work for ordered filtering. It would be good to add them as well. I suggest adding the `preserveFilterOrder` argument to `FileSystemGlobbingTestContext` and make an `OrderedPatternMatchingTests` derive from `PatternMatchingTests` so XUnit can pick up the same tests in the derived class.